File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/StudentTrainingLinkResponseDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.LinkType;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StudentTrainingLinkResponseDTO {
    private Long id;
    private Long studentId;
    private Long trainingId;
    private LinkType type;
    private LocalDateTime createdAt;
}

File: OrienGO-Frontend/src/app/model/dto/StudentTrainingLinkResponse.dto.ts
// src/app/model/dto/student-training-link-response.dto.ts

import { LinkType } from "../enum/LinkType.enum";

export interface StudentTrainingLinkResponseDto {
  id: number;
  studentId: number;
  trainingId: number;
  type: LinkType;
  createdAt?: string; // ISO date string
}
